zerver/views/realm.py

```diff
@@ -352,9 +354,32 @@ def update_realm(
 
 @require_realm_owner
 @has_request_variables
-def deactivate_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
+def deactivate_realm(
+    request: HttpRequest,
+    user: UserProfile,
+    realm_deletion_in_minutes: Optional[int] = REQ(
+        json_validator=check_none_or(check_int), default=None
+    ),
+) -> HttpResponse:
+    if (
+        realm_deletion_in_minutes is None
+        and settings.MAX_DEACTIVATED_REALM_DELETION_DAYS is not None
+    ):
+        raise JsonableError(_("Given data deletion time is not allowed."))
```

Comment:
We might want a more specific error here -- something about `Realms must provide a realm data deletion timeline of {__} or fewer days`